pkg/api/api.go

```diff
@@ -122,7 +124,27 @@ func (dvol *DvolAPI) setActiveVolume(volumeName string) error {
 	}
 	defer file.Close()
 	encoder := json.NewEncoder(file)
-	encoder.Encode(currentVolumeContent)
+	if err := encoder.Encode(currentVolumeContent); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (dvol *DvolAPI) setActiveBranch(volumeName, branchName string) error {
+	volume := dvol.dl.VolumeFromName(volumeName)
+	currentBranchJsonPath := filepath.FromSlash(volume.Path + "/current_branch.json")
+	currentBranchContent := map[string]string{
+		"current_branch": branchName,
+	}
+	file, err := os.Create(currentBranchJsonPath)
```

Comment:
Well the contract of Create is that it creates if it doesn't exist and opens and truncates if it does - I think this is what we want.